src/sparseml/transformers/sparsification/obcq/export.py

```diff
@@ -0,0 +1,634 @@
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Helper functions and script for exporting inference artifacts
+to be used by inference engines such as DeepSparse.
+The export incorporates:
+- creating the deployment directory (the direct input to the DeepSparse
+    inference pipeline)
+- creating an ONNX file representing a trained transformers model
+
+script accessible from sparseml.transformers.export_onnx_refactor
+
+command help:
+usage: sparseml.transformers.export_onnx [-h] --task TASK --model_path
+                                         MODEL_PATH
+                                         [--sequence_length SEQUENCE_LENGTH]
+                                         [--no_convert_qat]
+                                         [--onnx_file_name ONNX_FILE_NAME]
+                                         [--num_exported_samples NUM_EXPORTED_SAMPLES]
+                                         [--data_args DATA_ARGS]
+
+Export a trained transformers model to an ONNX file
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --task TASK           Task to create the model for. i.e. mlm, qa, glue, ner
+  --model_path MODEL_PATH
+                        Path to directory where model files for weights,
+                        config, and tokenizer are stored
+  --sequence_length SEQUENCE_LENGTH
+                        Sequence length to use. Default is 384. Can be
+                        overwritten later
+  --no_convert_qat      Set flag to not perform QAT to fully quantized
+                        conversion after export
+  --onnx_file_name ONNX_FILE_NAME
+                        Name for exported ONNX file in the model directory.
+                        Default and recommended value for pipeline
+                        compatibility is model.onnx
+  --num_export_samples NUM_EXPORT_SAMPLES
+                        Number of samples (inputs/outputs) to export
+  --data_args DATA_ARGS
+                        Valid json loadable args used to instantiate a
+                        `DataTrainingArguments` instance while exporting
+                        samples
+
+example usage:
+sparseml.transformers.export_onnx_refactor \
+  --task text_generation \
+  --model_path /PATH/TO/SPARSIFIED/MODEL/DIRECTORY \
+  --sequence_length 128
+"""
+
+import argparse
+import collections
+import copy
+import inspect
+import logging
+import os
+import shutil
+from typing import Any, Dict, List, Optional, Union
+
+import torch
+from torch.nn import Module
+from transformers import AutoConfig, AutoTokenizer
+from transformers.tokenization_utils_base import PaddingStrategy
+
+import sparseml.core.session as sml
```

Comment:
nit: we generally don't use import aliases, I think in this case it makes sense to alias, but maybe standardize to something a bit more explicit for readability (especially when splashing into other repos)